### PR TITLE
Add a new view for displaying awarded points in a modal dialog.

### PR DESCRIPTION
**Pull request by greinard**
## Overview

This branch adds a `NewPointsView` that can be used to display information about points earned in a modal dialog.

&nbsp;&nbsp;

Most of the content was specific to this new view, but I did split out the `AnimatedRing` as a separate component that can be reused.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing

- [x] MyDataHelps iOS app
- [x] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation

- I have added some entries in the Storybook to show how the new components can be configured.

### CODE REVIEW

**Comment by greinard** [comment]
I'd like to merge this PR fairly soon if someone has the time to give it a review.  Thank you.

**Comment by greinard** [comment]
Bumping in case anyone has a chance to review.  Not urgent, but it would be nice to close this out this week.

File: src/components/presentational/AnimatedRing/AnimatedRing.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { CSSProperties } from 'react'
+import './AnimatedRing.css'
+
+export interface AnimatedRingProps {
+    children: React.ReactNode;
+    style?: CSSProperties;
+    color?: string;
```
Comment (chrisnowak) [inline src/components/presentational/AnimatedRing/AnimatedRing.tsx:7]: Ideally use ColorDefinition here (which was added recently), but not a big deal

File: src/components/presentational/AnimatedRing/AnimatedRing.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export default function (props: AnimatedRingProps) {
```
Comment (chrisnowak) [inline src/components/presentational/AnimatedRing/AnimatedRing.tsx:10]: We have used circular progress bars in a number of components...so wondering if we should instead:
- Call this "ProgressRing"
- Have it take a % fill
- Have an "animated" property

Could see that being useful more broadly

File: src/helpers/strings-en.ts [hunk 3]
```diff
@@ -118,6 +118,11 @@
     "select-symptoms": "Select Symptoms",
     "include-treatments": "Include Treatments",
     "select-treatments": "Select Treatments",
+    "new-points-title": "Well Done!",
+    "new-points-text": "You've been awarded points for the following:",
+    "new-points-next-reward-prefix": "You now need ",
+    "new-points-next-reward-suffix": " points to unlock your next reward.",
+    "new-points-done-button-text": "DONE"
```
Comment (chrisnowak) [inline src/helpers/strings-en.ts:125]: I think for default out of box we should go with "Done" since we don't do all caps for any other buttons

**Review by chrisnowak — APPROVED** [review]
Conceptually makes sense and is a nice model for how we can generalize interactions that launch a view with parameters

File: src/components/presentational/AnimatedRing/AnimatedRing.tsx (hunk 1 above)
Comment (greinard) [reply src/components/presentational/AnimatedRing/AnimatedRing.tsx]: Done.

File: src/components/presentational/AnimatedRing/AnimatedRing.tsx (hunk 2 above)
Comment (greinard) [reply src/components/presentational/AnimatedRing/AnimatedRing.tsx]: Done.

File: src/helpers/strings-en.ts (hunk 3 above)
Comment (greinard) [reply src/helpers/strings-en.ts]: Done.

**Pull request merged**